src/transformation/utils/lua-ast.ts

```diff
@@ -22,24 +22,41 @@ export function createExportsIdentifier(): lua.Identifier {
     return lua.createIdentifier("____exports");
 }
 
-export function createExpressionPlusOne(expression: lua.Expression): lua.Expression {
-    if (lua.isNumericLiteral(expression)) {
-        const newNode = lua.cloneNode(expression);
-        newNode.value += 1;
+export const createExpressionPlusOne = (expression: lua.Expression) => modifyNumericExpression(expression, 1);
+export function modifyNumericExpression(expression: lua.Expression, change: number): lua.Expression {
```

Comment:
`modifyNumericExpression` is a little generic since this only adds/subtracts. Suggestions: `addToNumericExpression` or `addLiteralToNumericExpression`.